### ISSUE TITLE
0.39.0 docker image does not run

### ISSUE THREAD

**Issue by robertsben** [open-telemetry/opentelemetry-collector-contrib#6343]
**Describe the bug**
There is a permissions issue for the executable that has been shipped in 0.39.0 of the opentelemetry-collector-contrib docker image.

```
$ docker run otel/opentelemetry-collector-contrib:v0.39.0
Unable to find image 'otel/opentelemetry-collector-contrib:v0.39.0' locally
v0.39.0: Pulling from otel/opentelemetry-collector-contrib
1f66f23e2377: Already exists 
c134b4b489b6: Already exists 
Digest: sha256:a11e5a60bf7b987df7287f62a2e4efefa93cd023c1d277ab115e018bcbd054a5
Status: Downloaded newer image for otel/opentelemetry-collector-contrib:v0.39.0
docker: Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "/otelcontribcol": permission denied: unknown.
```

**Steps to reproduce**
```docker run otel/opentelemetry-collector-contrib:v0.39.0```

**What did you expect to see?**
The otelcontribcol executable should run normally. Below there is no config found, but this shows that the executable runs without permission errors.
```$ docker run otel/opentelemetry-collector-contrib:0.38.0
Unable to find image 'otel/opentelemetry-collector-contrib:0.38.0' locally
0.38.0: Pulling from otel/opentelemetry-collector-contrib
Digest: sha256:174c2f648e65d9bf4e6bc84aa0465400e4b4d9417c7b097cdef89b126ba9ed30
Status: Downloaded newer image for otel/opentelemetry-collector-contrib:0.38.0
Error: cannot load configuration's parser: error loading config file "/etc/otel/config.yaml": unable to read the file /etc/otel/config.yaml: open /etc/otel/config.yaml: no such file or directory
2021/11/16 12:15:03 collector server run finished with error: cannot load configuration's parser: error loading config file "/etc/otel/config.yaml": unable to read the file /etc/otel/config.yaml: open /etc/otel/config.yaml: no such file or directory
```

**What did you see instead?**
A clear and concise description of what you saw instead.

**What version did you use?**
`v0.39.0`

**What config did you use?**
N/A

**Environment**
MacOS 11.6.1, intel

```$ docker --version
Docker version 20.10.8, build 3967b7d
```

**Additional context**

The tag for the docker image has also deviated from the usual format e.g. `v0.39.0` instead of `0.39.0`

**Comment by mx-psi** [comment]
I can reproduce this (same macOS and Docker version, though I doubt that matters). cc @codeboten since you did last release

**Comment by indrekj** [comment]
It seems to be because otelcontribcol is not executable. My findings so far: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6238#issuecomment-970192993

**Comment by philipcwhite** [comment]
I have the same issue in GKE with kubernetes.

**Comment by codeboten** [comment]
Investigating now, this was the first release to be pushed using the github action instead of the circle ci job.

**Comment by codeboten** [comment]
Confirmed the issue is caused by the step which is downloading the binaries https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/3c2ca7d46b4fd656160dbb1ad359cb789a0367a4/.github/workflows/build-and-test.yml#L428

The fix in #6352 ensures the executable flag is set. We should add a step to validate the image before publishing it, will open an issue.
